a10_octavia/controller/worker/tasks/a10_network_tasks.py

```diff
@@ -684,10 +684,22 @@ def __init__(self, *arg, **kwargs):
 
     def _add_vrid_to_list(self, vrid_list, subnet, owner):
         vrid_value = CONF.a10_global.vrid
-        filtered_vrid_list = list(filter(lambda x: x.subnet_id == subnet.id, vrid_list))
-        if not filtered_vrid_list:
-            vrid_list.append(
-                data_models.VRID(
+        if isinstance(subnet, list):
+            subnet_ids = set([s.id for s in subnet])
+            for subnet_id in subnet_ids:
+                filtered_vrid_list = list(filter(lambda x: x.subnet_id == subnet_id, vrid_list))
```

Comment:
Is it possible to move line 690~698 to a function, and reuse it after in else case?

Reply:
Refactored this in https://github.com/a10networks/a10-octavia/pull/524